framework/db/Query.php

```diff
@@ -844,6 +849,119 @@ public function addParams($params)
     }
 
     /**
+     * @var array aliases defined for tables used in this query. `tablename => alias`.
```

Comment:
If we can not deal with multiple aliases, why we can not limit ourselves with applying alias to `ON` at `JOIN`, and leave everything else to developer?